osf/models/collection.py

```diff
@@ -31,11 +37,43 @@ class Meta:
     def _id(self):
         return self.guid._id
 
+    @cached_property
+    def _es_doc_id(self):
```

Comment:
Convention usually dictates that `_id` the same as what gets passed in to `load` -- you may want to update `_id`'s behavior to this instead.